jcl/src/openj9.dtfjview/share/classes/com/ibm/jvm/dtfjview/commands/OpenCommand.java

```diff
@@ -210,18 +212,29 @@ private void createContexts(Image loadedImage, String coreFilePath) {
 				}
 			}
 		}
+
 		if (!hasContexts) {
 			if (ctx.hasPropertyBeenSet(VERBOSE_MODE_PROPERTY)) {
 				out.println("Warning: no contexts were found, is this a valid core file?");
 			}
+		} else if (loadedImage.isTruncated()) {
+			out.println("Warning: dump file is truncated. Extracted information may be incomplete.");
+			out.println();
 		} else {
-			if (loadedImage.isTruncated()) {
-				out.println("Warning: dump file is truncated. Extracted information may be incomplete.");
-				out.println();
+			// check compatibility of this VM with the VM that created the core file
+			String thisVmName = System.getProperty("java.vm.name");
+			String coreVmName = ctx.getProperties().getProperty("java.vm.name");
+
+			if (thisVmName != null && coreVmName != null && !thisVmName.equals(coreVmName)) {
```

Comment:
I'm thinking it could be useful to print the warning even of the names is null.

Reply:
Sure, I can do that.